### PR TITLE
fix(LoaderStrategy): add `.ts` to allow for ts-node usage

### LINKED ISSUE #86: request: include `.ts` files in LoaderStrategy
Please include `.ts` files - support in `LoaderStrategy`.

### PR DESCRIPTION
**Pull request by Favna**
fixes #86

### CODE REVIEW

File: src/lib/strategies/LoaderStrategy.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { classExtends, isClass } from './Shared';
  */
 export class LoaderStrategy<T extends Piece> implements ILoaderStrategy<T> {
 	private readonly clientUsesESModules: boolean = getRootData().type === 'ESM';
-	private readonly supportedExtensions: readonly string[] = ['.js', '.cjs', '.mjs'];
+	private readonly supportedExtensions: readonly string[] = ['.js', '.cjs', '.mjs', '.ts'];
```
Comment (kyranet) [inline src/lib/strategies/LoaderStrategy.ts:17]: Drop `readonly`s, make field public, and add `.ts` only when `process.argv0` is `node-ts` (or contains it).

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/lib/strategies/LoaderStrategy.ts (hunk 1 above)
Comment (Favna) [reply src/lib/strategies/LoaderStrategy.ts]: Good shout

**Comment by Favna** [comment]
@kyranet didn't cross my mind lmao

File: src/lib/strategies/LoaderStrategy.ts [hunk 2]
```diff
@@ -13,8 +13,14 @@ import { classExtends, isClass } from './Shared';
  * Modules and CommonJS with reloading support.
  */
 export class LoaderStrategy<T extends Piece> implements ILoaderStrategy<T> {
-	private readonly clientUsesESModules: boolean = getRootData().type === 'ESM';
-	private readonly supportedExtensions: readonly string[] = ['.js', '.cjs', '.mjs'];
+	public clientUsesESModules: boolean = getRootData().type === 'ESM';
+	public supportedExtensions: string[] = ['.js', '.cjs', '.mjs'];
```
Comment (kyranet) [inline src/lib/strategies/LoaderStrategy.ts:17]:
```suggestion
	public clientUsesESModules = getRootData().type === 'ESM';
	public supportedExtensions = ['.js', '.cjs', '.mjs'];
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

File: src/lib/strategies/LoaderStrategy.ts [hunk 3]
```diff
@@ -13,8 +13,22 @@ import { classExtends, isClass } from './Shared';
...
  */
 export class LoaderStrategy<T extends Piece> implements ILoaderStrategy<T> {
-	private readonly clientUsesESModules: boolean = getRootData().type === 'ESM';
-	private readonly supportedExtensions: readonly string[] = ['.js', '.cjs', '.mjs'];
+	public clientUsesESModules = getRootData().type === 'ESM';
+	public supportedExtensions = ['.js', '.cjs', '.mjs'];
+
+	public constructor() {
+		/**
+		 * if {@linkplain https://github.com/TypeStrong/ts-node ts-node} is being used
```
Comment (kyranet) [inline src/lib/strategies/LoaderStrategy.ts:21]:
```suggestion
		 * If {@linkplain https://github.com/TypeStrong/ts-node ts-node} is being used
```

**Review by kyranet — CHANGES_REQUESTED** [review]
It's not a kyra LGTM if it's not approved -> requested -> approved

**Review by kyranet — APPROVED** [review]

**Pull request merged**